mijia/examples/fix-clock.rs

```diff
@@ -0,0 +1,142 @@
+//! Example to scan for sensors, connect to each one in turn, and correct its clock if necessary.
+
+use backoff::future::retry;
+use backoff::ExponentialBackoff;
+use chrono::{DateTime, Utc};
+use eyre::Report;
+use fmt::Write;
+use futures::TryFutureExt;
+use mijia::{MijiaSession, SensorProps};
+use std::fmt::{self, Debug, Formatter};
+use std::time::{Duration, SystemTimeError};
+use std::{process::exit, time::SystemTime};
+use tokio::time;
+use tokio_compat_02::FutureExt;
+
+const SCAN_DURATION: Duration = Duration::from_secs(5);
+/// Only correct clocks which are wrong by more than this amount.
+const MINIMUM_OFFSET: Duration = Duration::from_secs(10);
+const CONNECT_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// A duration which may be negative.
+#[derive(Clone, Eq, PartialEq)]
+struct SignedDuration {
```

Comment:
feels like you want something like https://docs.rs/utctimestamp/0.1.4/utctimestamp/struct.TimeDelta.html

Reply:
turns out we're already depending on chrono, so this wouldn't explode our dependency tree too much.

Reply:
It looks about right in intent, but doesn't have the conversion from `Result<Duration, SystemTimeError>` that is the main point.